clouddrift/binning.py

```diff
@@ -0,0 +1,177 @@
+"""Module for binning Lagrangian data."""
+
+from collections.abc import Iterable
+
+import numpy as np
+import xarray as xr
+
+DEFAULT_BINS_NUMBER = 10
+
+
+def binned_statistics(
+    coords: np.ndarray | list[np.ndarray],
+    data: np.ndarray | list[np.ndarray] | None = None,
+    bins: int | list = DEFAULT_BINS_NUMBER,
+    bins_range: list | None = None,
+    dim_names: list[str] | None = None,
+    output_names: list[str] | None = None,
+    zeros_to_nan: bool = False,
+) -> xr.Dataset:
+    """
+    Compute N-dimensional histogram binning and calculate variable means in each bin.
```

Comment:
Perform N-dimensional binning and compute mean of values in each bin. The result is returned as an Xarray Dataset.